roles/server/tasks/nginx.yml

```diff
@@ -0,0 +1,46 @@
+---
+- name: install nginx
+  apt: name=nginx
+
+- name: create nginx ssl directory
+  file:
+    path=/etc/nginx/ssl
+    state=directory
+
+- name: generate openssl key and crt
+  command: openssl req -new -newkey rsa:4096 -days 3650 -nodes -x509 -subj "/C=CA/ST=British Columbia/L=Vancouver/O=Korobi/OU=Web Development/CN={{ host }}" -keyout /etc/nginx/ssl/{{ site_name }}.key -out /etc/nginx/ssl/{{ site_name }}.crt
+  args:
+    creates: '/etc/nginx/ssl/{{ site_name }}.crt'
```

Comment:
It'd be neat if we could add this certificate to the system's trust store - would make it no longer necessary to add it to the browser's exception list and tools like curl etc would work when talking to the VM. Perhaps that could be achieved with a [local playbook](http://docs.ansible.com/ansible/playbooks_delegation.html#local-playbooks) file/task?